git_test.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+func TestGit(t *testing.T) {
+	workingDir, err := os.Getwd()
+	if err != nil {
+		t.Error()
+	}
+
+	scm := &GitScm{
+		repositoryPath: workingDir,
+	}
+
+	atts := scm.contributeOtelAttributes()
+
+	assert.Equal(t, 3, len(atts))
+
+	assert.Condition(t, func() bool { return keyExists(t, atts, ScmBranch) }, "Branch is not set as scm.branch")
```

Comment:
Only checking that the key exists, because, without doing some mocking, the execution of the test will be coupled with the current git state (i.e. a local branch for development purpose)